src/start-stop-daemon/start-stop-daemon.c

```diff
@@ -21,10 +21,8 @@
 
 #define ONE_MS           1000000
 
-#ifdef __linux__
-/* For extra SCHED_* defines. */
+/* For extra SCHED_* defines and pipe2. */
```

Comment:
the pipe2 usage implies a bigger problem here i think.  we're using it unprotected.  we really need a test in meson to see if the system supports pipe2 and switch between it & standard pipe.

Reply:
while this is correct, OTOH it seems that `pipe2()` is available in all the OSes for which there is some kind of support in OpenRC:
- Linux & Hurd: provided by glibc
- FreeBSD: https://man.freebsd.org/cgi/man.cgi?query=pipe2&sektion=2&n=1
- OpenBSD: https://man.openbsd.org/pipe.2
- NetBSD: https://man.netbsd.org/pipe2.2
- DragonFlyBSD: https://man.bsd.lv/DragonFly-5.6.1/man2/pipe2.2

so adding a meson test for it with fallback to `pipe()` would effectively create dead code, because it would not be used on any of the supported OSes; considering that currently `pipe2()` is used unconditionally without issues, IMHO it would be acceptable to keep using it like that, and wait until anyone tries to port OpenRC to another OS

Reply:
> see if the system supports pipe2 and switch between it & standard pipe.

I haven't taken a look into the usage yet, so this might not be a problem, but `pipe2` and `pipe + fcntl` aren't functionally equivalent.

With `pipe + fcntl` there exists a window between these two calls where the `fd` might not have proper mode set. This can become an issue in multi-threaded or signal handler contexts.